core/lib/src/config/builder.rs

```diff
@@ -314,8 +314,9 @@ impl ConfigBuilder {
     /// ```
     pub fn finalize(self) -> Result<Config> {
         let mut config = Config::new(self.environment);
+
         config.set_address(self.address)?;
-        config.set_port(self.port);
+        config.set_port(self.port.value());
```

Comment:
```suggestion
        config.set_port(*self.port);
```

(Assuming that works...)